task/acs-deploy-check/0.1/acs-deploy-check.yaml

```diff
@@ -0,0 +1,82 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: acs-deploy-check
+spec:
+  description: >-
+    Policy check a deployment with StackRox/RHACS This tasks allows you to check
+    a deployment against build-time policies and apply enforcement to fail
+    builds. It's a companion to the stackrox-image-scan task, which returns full
+    vulnerability scan results for an image.
+  params:
+    - name: rox-central-endpoint
+      type: string
+      description: |
+        The address:port tuple for StackRox Central.
+        For example: rox.stackrox.io:443
+    - name: rox-api-token-secret-name
+      type: string
+      description: |
+        Secret containing the StackRox API token with CI permissions under rox-api-token key
+    - name: verbose
+      type: string
+      default: 'true'
+    - name: insecure-skip-tls-verify
+      type: string
+      default: 'false'
+      description: |
+        When set to `"true"`, skip verifying the TLS certs of the Central
+        endpoint.  Defaults to `"false"`.
+  results:
+    - description: Output of `roxctl deployment check`
+      name: check_output
+  volumes:
+    - name: repository
+      emptyDir: {}
+    - name: rox-secret
+      secret:
+        secretName: $(params.rox-api-token-secret-name)
+        optional: true
+  steps:
+    - name: rox-deploy-scan
+      image: 'registry.access.redhat.com/ubi8:8.7-1026'
+      volumeMounts:
+        - name: repository
+          mountPath: /workspace/repository
+        - name: rox-secret
+          mountPath: /rox-secret
+      workingDir: /workspace/repository
+      script: |
+        #!/usr/bin/env bash
+
+        # Check if rox API token is configured
+        if test -f /rox-secret/rox-api-token ; then
+          export ROX_API_TOKEN=$(</rox-secret/rox-api-token)
+        else
+          echo "rox API token is not set, demo will exit with success"
+          echo "TODO: configure the pipeline to have access to ROXCTL"
+          exit 0
+        fi
+
+        echo "Using rox central endpoint $(params.rox-central-endpoint)"
+
+        echo "Configuration"
+        echo "Directory = $(pwd)"
+        ls -al
+
+        curl -s -k -L -H "Authorization: Bearer $ROX_API_TOKEN" \
+          "https://$(params.rox-central-endpoint)/api/cli/download/roxctl-linux" \
```

Comment:
NACK on this. This effectively allows users to execute an arbitrary binary in this task. This is effectively introduces a variation of the issue https://issues.redhat.com//browse/STONEBLD-2109.

Reply:
The "bring your own ACS" which is off-cluster is a use case we need to support.

Reply:
I understand the requirement, but this can't be the only possible way to achieve it.

Is ACS on the same cluster? If so, you can assume a certain URL, no?

As a stop gap, we need to, at a bare minimum, add an EC policy that verifies the value of the `rox-central-endopint` parameter is one from a list of pre-authorized values (which can be customized.)

Reply:
Would it be preferable to have that validation in EC or within the task itself? In-task validation seems more desirable for this use case than some others that have cropped up.

Reply:
Unclear what the best approach is here. It comes down to do we trust the task or do we only trust the task when used in a certain way.